04.ls/ls.rb

```diff
@@ -57,4 +95,79 @@ def puts_table(entry_name_table, max_width)
   end
 end
 
+def create_entry_info(entry_names)
+  total_blocks = 0
+  entry_info_table = entry_names.map do |entry_name|
+    entry_info = {}
+    stat = File::Stat.new(entry_name)
+    total_blocks += stat.blocks
+    entry_info[:file_mode] = get_file_mode(stat, entry_name)
+    entry_info[:nlink] = stat.nlink.to_s
+    entry_info[:owner] = get_owner(stat)
+    entry_info[:group_name] = get_group_name(stat)
+    entry_info[:size] = stat.size.to_s
+    updated_day = File.mtime(entry_name)
+    entry_info[:updated_date] = get_updated_date(updated_day)
+    entry_info[:name] = entry_name
+    entry_info
+  end
+  [total_blocks, entry_info_table]
+end
+
+def get_file_mode(stat, entry_name)
+  stat_mode = stat.mode.to_s(8)
+  aligned_stat_mode = stat_mode.rjust(6, '0')
+  file_type = FILE_TYPES[aligned_stat_mode[0, 2]]
+  file_permission = get_file_permission(aligned_stat_mode)
+  extended_attributes = get_extended_attributes(entry_name)
+  "#{file_type}#{file_permission}#{extended_attributes}"
+end
+
+def get_file_permission(stat_mode)
+  stat_mode[3..5].chars.map { |digit| FILE_PERMISSION_TABLES[digit] }.join
+end
+
+def get_extended_attributes(file_name)
+  attrs = `xattr #{file_name}`
+  attrs.empty? ? ' ' : '@'
+end
+
+def get_owner(stat)
+  Etc.getpwuid(stat.uid).name
+end
+
+def get_group_name(stat)
+  Etc.getgrgid(stat.gid).name
+end
+
+def get_updated_date(updated_day)
```

Comment:
mtimeメソッドはTimeオブジェクトを返すので、timeと呼んであげましょう。
また、人間が読みやすいように整形する処理をformatと呼ぶので、メソッド名はこんな感じがいいかも。

```suggestion
def format_time(time)
```

Reply:
format_timeに修正しました。
メソッド名の修正に伴い、メソッド内の処理やkeyもupdated_date→timeとなるように修正しました。